crates/gitbutler-core/src/git/commit_buffer.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::Result;
+use core::str;
+
+pub struct CommitBuffer {
+    heading: Vec<(String, String)>,
+    message: String,
+}
+
+impl CommitBuffer {
+    pub fn new(buffer: &[u8]) -> Result<Self> {
```

Comment:
It's more common for `new()` to be infallible, like a 'dumb' constructor. Since this is parsing the bytes buffer, ideally there is a name that indicates more work is done, like `from_bytes()` (used in `gix`, or `decode`).